substrate/frame/assets/src/lib.rs

```diff
@@ -1798,6 +1840,105 @@ pub mod pallet {
 		}
+
+		/// Revoke irreversibly Owner, Issuer, Admin, and Freezer privileges of the asset. Freeze
+		/// the metadata.
+		///
+		/// Warning: This action is irreversible. Reinstating privileged roles would require an
+		/// action from `ForceOrigin`.
+		///
+		/// Origin must be either Signed and the sender should be the Owner of the asset `id` or
+		/// `ForceOrigin`.
+		///
+		/// If origin is the owner: The deposit of the asset details and metadata will be
+		/// unreserved then withdrawn and sent to `DepositDestinationOnRevocation`. If no metadata
+		/// was set then the new deposit for the new metadata will be withdrawn from the owner.
+		///
+		/// If origin is `ForceOrigin`: The deposit of owner is unreserved.
+		///
+		/// Asset must be in `Live` status.
+		///
+		/// - `id`: The identifier of the asset.
+		///
+		/// Emits `OwnerAndTeamRevoked` and `MetadataSet`.
+		///
+		/// Weight: `O(1)`
+		#[pallet::call_index(33)]
+		pub fn revoke_all_privileges(
+			origin: OriginFor<T>,
+			id: T::AssetIdParameter,
+		) -> DispatchResult {
+			let maybe_owner_origin = match T::ForceOrigin::try_origin(origin) {
+				Ok(_) => None,
+				Err(origin) => Some(ensure_signed(origin)?),
+			};
+			let id: T::AssetId = id.into();
+			let mut asset = Asset::<T, I>::get(id.clone()).ok_or(Error::<T, I>::Unknown)?;
+
+			if let Some(owner) = maybe_owner_origin.as_ref() {
+				ensure!(asset.owner() == Some(&owner), Error::<T, I>::NoPermission);
+			}
+			ensure!(asset.status == AssetStatus::Live, Error::<T, I>::IncorrectStatus);
+
+			let old_metadata = Metadata::<T, I>::get(&id);
+
+			let deposit_amount = asset.deposit.saturating_add(old_metadata.deposit);
+
+			let fee_imbalance = if let Some(owner) = maybe_owner_origin {
+				let new_metadata_deposit =
+					Self::calc_metadata_deposit(&old_metadata.name, &old_metadata.symbol);
+
+				// NOTE: if new deposit is less than old deposit then the excess is not refunded.
+				let metadata_deposit_diff =
+					new_metadata_deposit.saturating_sub(old_metadata.deposit);
+
+				with_storage_layer::<_, DispatchError, _>(|| {
+					T::Currency::unreserve(&owner, deposit_amount);
+					let imbalance = T::Currency::withdraw(
+						&owner,
+						deposit_amount.saturating_add(metadata_deposit_diff),
+						WithdrawReasons::FEE,
+						ExistenceRequirement::KeepAlive,
+					)?;
+					Ok(imbalance)
+				})?
+			} else if let Some(owner) = asset.owner() {
+				T::Currency::unreserve(owner, deposit_amount);
+				Imbalance::zero()
+			} else {
+				log::error!(
+					target: LOG_TARGET,
+					"asset is ensured to be `Live` thus asset has an owner; qed"
+				);
+				Imbalance::zero()
+			};
+
+			let new_metadata = AssetMetadata {
+				deposit: Zero::zero(),
+				name: old_metadata.name,
+				symbol: old_metadata.symbol,
+				decimals: old_metadata.decimals,
+				is_frozen: true,
+			};
+			Metadata::<T, I>::insert(&id, &new_metadata);
+
+			asset.deposit = Zero::zero();
+			asset.status = AssetStatus::LiveAndNoPrivileges;
```

Comment:
AFAIK we do not have invalid account address for our crypto. not sure how reasonable (probably not) would be to have some valid but random address to be used as a common null address.

Reply:
It is indeed a bit safer in case of error in the runtime.

We can copy the logic of pure proxy.

Reply:
thinking again, the best would be a special account owned by the asset pallet (same as the treasury account), which doesn't have any action.

I will do it.

Reply:
done in https://github.com/paritytech/polkadot-sdk/pull/4150/commits/5d30a4ee040af15cdb84fa7dc96651cdf1919d27

I copied the pallet proxy logic.